src/backend/distributed/executor/intermediate_results.c

```diff
@@ -940,6 +943,16 @@ fetch_intermediate_results(PG_FUNCTION_ARGS)
 static uint64
 FetchRemoteIntermediateResult(MultiConnection *connection, char *resultId)
 {
+	char *localPath = QueryResultFileName(resultId);
```

Comment:
Is it possible that we fetch the same intermediate result more than once?

Reply:
The skip-if-file-exists is mainly to catch fetches to self. Previously, different names were used between sender and receiver. Intermediate results currently have the same name everywhere.

We do not repeat/retry fetches, though we probably need to reconsider this logic if we do, since we may only have a partial file. The "simplest" solution would be to prune away fetch-to-self tasks, after which we can remove the skip-if-file-exists logic below.